src/ngx-smart-modal/src/services/ngx-smart-modal-stack.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Injectable } from '@angular/core';
+
+import { NgxSmartModalComponent } from '../../src/components/ngx-smart-modal.component';
+import { ModalInstance } from './modal-instance';
+
+@Injectable()
+export class NgxSmartModalStackService {
```

Comment:
I have some little questions about this service because I think it shouldn't be exposed publicly. It gives services to the main service but we don't have to propose two similar methods from different sources. 😉